loadable_apps/loadable_sample/wifiapp/binary_update.c

```diff
@@ -64,49 +66,47 @@ static void binary_update_cb(void)
 	printf(" ========================================================================= \n");
 }
 
-static int binary_update_download_binary(binary_update_info_t *binary_info, int condition)
+static int binary_update_download_binary(char *origin_path, binary_update_info_t *binary_info, int condition)
 {
 	int read_fd;
 	int write_fd;
-	int new_version;
 	int ret;
 	int total_size;
 	int copy_size;
 	int read_size;
+	uint32_t new_version;
 	uint32_t crc_hash = 0;
 	uint8_t buffer[BUFFER_SIZE];
+	char download_path[BINARY_PATH_LEN];
 	user_binary_header_t header_data;
-	char filepath[BINARY_PATH_LEN];
-	char new_filepath[BINARY_PATH_LEN] = {0,};
 
-	snprintf(filepath, BINARY_PATH_LEN, "%s/%s_%u", BINARY_DIR_PATH, binary_info->name, (uint32_t)binary_info->version);
-	read_fd = open(filepath, O_RDONLY);
+	read_fd = open(origin_path, O_RDONLY);
```

Comment:
if CONFIG_BCH is enabled, inode link it to bch.
SO now it may uses bch driver.
As I Know, bch driver allow first write, and append. It doesn't allow truncate to remove old data.
So I think this will not be work properly with some version of binaries.

Reply:
Thank you for your technical concerns.
I checked it is working well with new written data.
As I said before on messenger, this can be written by ftl_flush which erases blocks before writing new data internally.